### PR TITLE
Improvement of installation

### PR DESCRIPTION
**Pull request by gordonwisc**
1. Parse UUID to ncp-rc.yaml with overlay_tz, t0, container_up_block,
   snat_ip_pool.
2. preconfigured parameters, reduced parameters needs to be configured
3. Added openshift installation playbook. It's a playbook that automatically
   pulls and run openshift playbook. If openshift is already installed, it
   will restart master api and controller.
4. Added install playbook to run ncp_prep openshift installation and ncp playbook

### CODE REVIEW

File: nsx_config.py [hunk 1]
```diff
@@ -646,16 +647,18 @@ def _handle_general_configuration(self, resource_type, resource_name,
         if not self._has_tags(resource, required_tags):
```
Comment (kais66) [inline nsx_config.py]: remove line 647-650

File: openshift-ansible-nsx/hosts [hunk 2]
```diff
@@ -47,3 +47,5 @@ openshift_disable_check=disk_availability,memory_availability,docker_storage,pac
 openshift_hosted_manage_registry=false
 openshift_hosted_manage_router=false
 openshift_enable_service_catalog=false
+skip_sanity_checks=true
```
Comment (kais66) [inline openshift-ansible-nsx/hosts]: this has been addressed from upstream.

File: openshift-ansible-nsx/hosts [hunk 3]
```diff
@@ -47,3 +47,5 @@ openshift_disable_check=disk_availability,memory_availability,docker_storage,pac
...
+openshift_web_console_install=true
```
Comment (kais66) [inline openshift-ansible-nsx/hosts]: maybe we should set this to false?

File: openshift-ansible-nsx/roles/openshift_installation/defaults/main.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+---
+perform_openshift_installation: True
+host_path: /root/nsx-integration-for-openshift-test/openshift-ansible-nsx/hosts
```
Comment (kais66) [inline openshift-ansible-nsx/roles/openshift_installation/defaults/main.yaml]: it'd be better to use a relative path

File: openshift-ansible-nsx/roles/openshift_installation/tasks/main.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+- name: Get Openshift playbook from repo
+  git:
+      repo: "https://github.com/openshift/openshift-ansible.git"
+      dest: /tmp/openshift-ansible
+      version: release-3.9
+  when: perform_openshift_installation == True
+
+- name: print
```
Comment (kais66) [inline openshift-ansible-nsx/roles/openshift_installation/tasks/main.yaml]: remvoe

File: openshift-ansible-nsx/hosts (hunk 3 above)
Comment (gordonwisc) [reply openshift-ansible-nsx/hosts]: yes.  done

File: openshift-ansible-nsx/roles/openshift_installation/tasks/main.yaml (hunk 5 above)
Comment (gordonwisc) [reply openshift-ansible-nsx/roles/openshift_installation/tasks/main.yaml]: done

File: openshift-ansible-nsx/roles/openshift_installation/defaults/main.yaml (hunk 4 above)
Comment (gordonwisc) [reply openshift-ansible-nsx/roles/openshift_installation/defaults/main.yaml]: use ansible env variable {{ inventory_dir }} instead

File: nsx_config.py (hunk 1 above)
Comment (gordonwisc) [reply nsx_config.py]: done

**Comment by kais66** [comment]
LGTM

**Pull request merged**